Add support for embedded surveys in post or page content on webkit.org

jdatapple opened the pull request:
#### 0ee248acc11659aa7ba747c89638068a5e5116b9
<pre>
Add support for embedded surveys in post or page content on webkit.org
<a href="https://bugs.webkit.org/show_bug.cgi?id=249342">https://bugs.webkit.org/show_bug.cgi?id=249342</a>

Reviewed by Timothy Hatcher.

* Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php: Added.
* Websites/webkit.org/wp-content/plugins/webkit-survey/results.php: Added.
* Websites/webkit.org/wp-content/plugins/webkit-survey/survey.php: Added.

Canonical link: <a href="https://commits.webkit.org/257891@main">https://commits.webkit.org/257891@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e91f004be5c5d8a51df45f9220f10bd0928f4ebb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/6/builds/100339 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/9501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/109657 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/10412 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/38 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/92740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/107535 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106117 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/10412 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/92740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/10412 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/92740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/3243 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/3236 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/38 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/9357 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/33403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/5052 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2801 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Websites/webkit.org/wp-content/plugins/webkit-survey/survey.php:
```diff
@@ -0,0 +1,95 @@
+<p class="button-align-right">
+    <input type="submit" name="Submit" value="Submit" class="submit-button">
+</p>
+</form>
```
Should add a newline at the end of the file.

jdatapple replied:
Addressed in 2619fb8

xeenon commented on Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php:
```diff
@@ -0,0 +1,224 @@
+    }
+}
+
+WebKit_Survey::init();
```
Newline here too.

jdatapple replied:
Addressed in 2619fb8

xeenon commented on Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php:
```diff
@@ -0,0 +1,224 @@
+        return ob_get_clean();
+    }
+
+    public function add_survey_boxes () {
```
No space before `()`

jdatapple replied:
Addressed in e91f004

jdatapple committed:
Add support for embedded surveys in post or page content on webkit.org
https://bugs.webkit.org/show_bug.cgi?id=249342

Reviewed by Timothy Hatcher.

* Websites/webkit.org/wp-content/plugins/webkit-survey/plugin.php: Added.
* Websites/webkit.org/wp-content/plugins/webkit-survey/results.php: Added.
* Websites/webkit.org/wp-content/plugins/webkit-survey/survey.php: Added.

Canonical link: https://commits.webkit.org/257891@main
